pyOCD/flash/flash_builder.py

```diff
@@ -573,8 +588,15 @@ def _page_erase_program_double_buffer(self, progress_cb=_stub_progress):
                 # Kick off this page program.
                 current_addr = page.addr
                 current_weight = page.getEraseProgramWeight()
+
+                self.flash.init(fnc=self.flash.INIT_FUNC_ERASE)
                 self.flash.erasePage(current_addr)
+                self.flash.unInit(self.flash.INIT_FUNC_ERASE)
+
+                self.flash.init(fnc=self.flash.INIT_FUNC_PROGRAM)
                 self.flash.startProgramPageWithBuffer(current_buf, current_addr) #, erase_page=True)
+                self.flash.unInit(self.flash.INIT_FUNC_PROGRAM)
```

Comment:
You'll need to wait for the program operation to complete before uninit is called. If you move uninit to after self.flash.waitForCompletion() on line 609 it should be good.

Reply:
fixed